x-pack/legacy/plugins/ml/public/jobs/new_job_new/pages/components/datafeed_step/datafeed.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, FC, useContext, useEffect, useState } from 'react';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { WizardNav } from '../wizard_nav';
+import { QueryInput } from './components/query';
+import { QueryDelayInput } from './components/query_delay';
+import { FrequencyInput } from './components/frequency';
+import { ScrollSizeInput } from './components/scroll_size';
+import { TimeField } from './components/time_field';
+import { WIZARD_STEPS, StepProps } from '../step_types';
+import { JobCreatorContext } from '../job_creator_context';
+import { JsonEditorFlyout, EDITOR_MODE } from '../common/json_editor_flyout';
+
+export const DatafeedStep: FC<StepProps> = ({ setCurrentStep, isCurrentStep }) => {
+  const { jobValidator, jobValidatorUpdated } = useContext(JobCreatorContext);
+  const [nextActive, setNextActive] = useState(false);
+  const [validQuery, setValidQuery] = useState(false);
```

Comment:
Nit: Maybe `isValidQuery` makes it more clear it's a `boolean`. Since it's passed on to another component and seeing it there I wasn't sure if it's a flag or if it contains a valid query.